### ISSUE TITLE
Imports that end with 'not' are not detected by the unused-import rule

### ISSUE THREAD

**Issue by fmordashev** [pinterest/ktlint#1587]
## Expected Behavior
While using the detekt plugin (https://github.com/detekt/detekt) unused imports should be removed, since my config file includes
```
  NoUnusedImports:
    active: true
    autoCorrect: true
```

As stated by the plugin developers, the unused-import rule is wrapped around the ktlint rule and I was advised to report the issue here.

## Observed Behavior
The import is not removed if it ends with 'not' text

## Steps to Reproduce
Can be easily tested with any import that ends with 'not' while running detekt/ktlint checks
For example:
```
package my.package.cool.printer

import my.package.cool.matchers.Matcher.not

class Printer() {

  fun printHello() {
     print("hello")
  }
}
```

Initially found for `import org.hamcrest.Matchers.not` from `org.hacrest:hamcrest:2.2`

## Your Environment

* Version of ktlint used: used detekt plugin of version 1.19.0 and 1.21.0, also tried using the ktlint-gradle plugin https://github.com/JLLeitschuh/ktlint-gradle (v 10.3.0)
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task): can't provide due to ssl restrictions
* Operating System and version: win10

**Comment by paul-dingemans** [comment]
Tnx for reporting. Actually this is a good find.

Any import ending with one of folllowing will suffer from the same problem:
```
        val operatorSet = setOf(
            // unary
            "unaryPlus", "unaryMinus", "not",
            // inc/dec
            "inc", "dec",
            // arithmetic
            "plus", "minus", "times", "div", "rem", "mod", "rangeTo",
            // in
            "contains",
            // indexed access
            "get", "set",
            // invoke
            "invoke",
            // augmented assignments
            "plusAssign", "minusAssign", "timesAssign", "divAssign", "modAssign",
            // (in)equality
            "equals",
            // comparison
            "compareTo",
            // iteration (https://github.com/shyiko/ktlint/issues/40)
            "iterator",
            // by (https://github.com/shyiko/ktlint/issues/54)
            "getValue", "setValue"
        )
```

The check regarding whether the last part of an an import path occurs in the operatorSet seems to be way to strict. It should only prohibit the removal of such an import if the corresponding operator (https://kotlinlang.org/docs/keyword-reference.html#operators-and-special-symbols) is actually used in the code. When your code does not contain the not/negate operator it should just mark the import as unused. Biggest concern with this rule is that it should not accidently remove imports that are actually used as that would lead to compilation failures.

Can you confirm that your original code indeed did not contain a not/negate operator?

**Comment by fmordashev** [comment]
Yes, I am 100% certain, there were no methods or operators that were named ‘not’ aside from the import statement

**Comment by ivanalvarado** [comment]
This might be related. I have an unused import that ends with `"string"` and it too is not being reported.

The import is
```
import com.esotericsoftware.kryo.util.Util.string
```
and when I run `ktlint` it **does not** get reported as unused. However, when I drop `"string"` and import by doing
```
import com.esotericsoftware.kryo.util.Util
```
it correctly **does** get reported as unused.

**Comment by paul-dingemans** [comment]
> This might be related. I have an unused import that ends with `"string"` and it too is not being reported.
> 
> The import is
> 
> ```
> import com.esotericsoftware.kryo.util.Util.string
> ```
> 
> and when I run `ktlint` it **does not** get reported as unused. However, when I drop `"string"` and import by doing
> 
> ```
> import com.esotericsoftware.kryo.util.Util
> ```
> 
> it correctly **does** get reported as unused.

Can you provide a code sample as I can not reproduce your finding:
```
 $ echo -e "package com.example\n\nimport com.esotericsoftware.kryo.util.Util.string\n\n" | ktlint-0.47.1 --stdin
16:53:01.357 [main] INFO com.pinterest.ktlint.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
<stdin>:3:1: Unused import (no-unused-imports)
```

**Comment by paul-dingemans** [comment]
I took a deep dive into this topic. It is going to be too complicated to be added. The psi does not contain enough information to deduct with certainty that an import is used or unused. As falsely removing a dependency results in a compilation error, it is better to accept that the rule can not detect all unused parameters.

I am open to contributions which do solve this problem in a smart way.
